CustomWeapons.cpp

```diff
@@ -235,6 +235,18 @@ HOOK_METHOD(ProjectileFactory, NumTargetsRequired, () -> int)
     LOG_HOOK("HOOK_METHOD -> ProjectileFactory::NumTargetsRequired -> Begin (CustomWeapons.cpp)\n")
     int ret = super();
 
+    if(blueprint->type == 2)
+    {
+        if(blueprint->length>1)
+        {
+            return 2; //fix for charge beams crashing the game
```

Comment:
Do charge beams work correctly when you fire them for different numbers of charges?

Reply:
The damage doesn't change normally and the animation plays out like a normal beam animation. However, you can modify the damage in lua so that any of the Damage attributes scale with the number of charges.